### ISSUE TITLE
Unable to add images from nextcloud

### ISSUE THREAD

**Issue by manu-torres** [CollaboraOnline/online#8226]
### Describe the Bug
Images can't be added from the nextcloud storage.

### Steps to Reproduce
1. Open any document
2. Try to insert an image stored on the nextcloud server
3. Select an image and accept
4. An error appears

![Captura de pantalla de 2024-02-08 12-42-31](https://github.com/CollaboraOnline/online/assets/56218303/28665b74-d0ee-48c0-b636-31892fbceb9b)

Images from the local computer (uploaded through the browser) work fine.

### Expected Behavior
Image should be added

### Actual Behavior
An error appears.

### Desktop
(Please complete the following information)

Problem appears on the nextcloud web interface.

- COOLWSD version: 23.05.7.5
- LOKit version: 23.05.7.5
- Served by: Debian GNU/Linux 12 (bookworm)
- Nextcloud version: 27.1.5
- Nextcloud AIO version: v7.11.2 (automatic updates enabled, all containers up to date)

Both Nextcloud and Collabora run on the same machine (managed by AIO).

### Additional Context
I am aware of this [issue](https://github.com/CollaboraOnline/online/issues/8129), which is marked as solved, but I am not sure it will fix my problem. They suggest to add the nextcloud server to the list of trusted sources in `/etc/coolwsd/coolwsd.xml`.

I installed collabora with Nextcloud AIO, using the AIO web interface (just by selecting the Collabora container), and in theory the nextcloud container should be in one of the subnets already in the trusted sources. ifconfig inside the nextcloud container reports nextcloud is on 172.18.0.8, which should fall under the 172.16.0.0/12 subnet already in trusted sources.

Perhaps it is not the case and I have to add another trusted source, in that case I have some questions:

- How can I find out the IP of the container running nextcloud (ifconfig inside the container?)
- How can I edit `/etc/coolwsd/coolwsd.xml` inside the container to add the address. There is no text editor installed, and even after installing one I can't edit the file as nano does not seem be working. I have permissions to edit the file but the terminal gets confused when inside de container.

The [release notes](https://www.collaboraoffice.com/collabora-online-23-05-release-notes/) from Collabora Online 23.05.7 (released on 18 January 2024) claim to have fixed the issue. I installed Collabora before that date but updated since.

Hope that I am not missing something important and that this is the right place to post.

**Comment by manu-torres** [comment]
Ok, now I know which host I have to add, but how can I add the host to the list then. I guess I have to edit the file `/etc/coolwsd/coolwsd.xml` inside the container, but it has been challenging. I tried to install a text editor but can't edit the file. Also tried to copy the file to the host machine (outside the container), add the line and then copy it back to the container with docker cp.

Neither of them have worked so far. I am not even sure the changes would be persisted, even if I managed to edit the file, chances  are they would be lost after reboot. This would be trivial if I installed on the server directly, but docker and AIO make it more puzzling. Maybe it's a problem with AIO not making the config properly rather than collabora?

**Comment by sebastiaanveld** [comment]
I ran into the same issue, see https://github.com/CollaboraOnline/online/issues/8129

Just copy the coolwsd.xml from the container to the host;

```
mkdir -p /data/local/etc/coolwsd
cd /data/local/etc/coolwsd
docker cp <containerprocess|containername>:/etc/coolwsd/coolwsd.xml .

```
Change the ownership of the file too id 100, this is the ID of the cool user in the container;
chown 100 coolwsd.xml

Now ajust coolwsd.xml so it lists you NC server; my coolwsd.xml shows my change as:
```
<host desc="Localhost access by name">192\.168\.250\.[0-9]{1,3}</host>
</lok_allow>
```

Start your COOL container with -v to mount the local coolwsd.xml to the container;
`-v /data/local/etc/coolwsd/coolwsd.xml:/etc/coolwsd/coolwsd.xml`

**Comment by manu-torres** [comment]
Hi, thanks for the response. I came across your issue while searching, but for some reason I did not think about "mounting" the file to the host. That will most likely fix the issue and let me add the line, but once again containers make things more difficult. Since I am using AIO I don't have the container deployment commands on the host, instead they can be found inside the `nextcloud-aio-mastercontainer`, on the file `/var/www/docker-aio/php/containers.json`.

I guess I have to mount that file as well, edit the collabora container definition in `containers.json` to allow me to then mount the other file with the `-v` option as you suggest on the collabora container. 

The thing is that in order to change `containers.json` I have to delete the `nextcloud-aio-mastercontainer` and create it again, and messing with this container has a (small) chance of nuking the entire instance. Since the error is fairly bening and can be circunvented by uploading the file from the browser, I am not sure if I want to tackle it at the moment.

Since the domain that I have to add to the list is just the domain of the server this should most lilkely happen automatically and this must be a bug with AIO not setting up the containers properly, rather than being a Collabora issue.

I will ask over the AIO but for now I am closing the issue.

Thanks everybody, I'll see if I figure something out.
